src/app/services/authentication.service.ts

```diff
@@ -1,36 +1,48 @@
 import { Injectable } from '@angular/core';
 import {HttpClient, HttpHeaders} from '@angular/common/http';
-import { Observable } from 'rxjs';
 import { AppService } from '../app.service';
+import { Observable, Subject } from 'rxjs';
 const httpOptions = {
   headers: new HttpHeaders(
   { 'Content-Type': 'application/json',
-    'Fineract-Platform-TenantId': 'mobile'})
+    'Fineract-Platform-TenantId': 'mobile',
+    'Access-Control-Allow-Methods': 'POST, GET, OPTIONS, DELETE, PUT',
+    'Access-Control-Allow-Origin': '*'})
 };
 @Injectable({
   providedIn: 'root'
 })
 export class AuthenticationService {
   userStatus = false;
-  key = 'token';
+  token = '';
+  flag = new Subject<boolean>();
   constructor(private http: HttpClient, private appService: AppService) { }
-   login(username: string, password: string) {
-     if (username === 'mifos' && password === 'password') {
-      localStorage.setItem(this.key, 'bWlmb3M6cGFzc3dvcmQ=');
-      this.userStatus = true;
-      this.appService.emitConfig(this.userStatus);
-      this.http.post<any>(
-        'https://mobile.openmf.org/fineract-provider/api/v1/authentication?username=mifos&password=password',
-        httpOptions).subscribe((data: any) => console.log(data));
-      return true;
-     } else {
-       return false;
-     }
-
-    // return this.http.post<any>(
-    //   'https://mobile.openmf.org/fineract-provider/api/v1/authentication?username=' + `${username}` + '&password=' + `${password}`,
-    //   httpOptions).subscribe((data: any) => console.log(data));
-    // }
+   login(username: string, password: string): Observable<boolean> {
+      this.http.post<any[]>
+      ('https://mobile.openmf.org/fineract-provider/api/v1/self/authentication?username=' + `${username}` + '&password=' + `${password}`,
```

Comment:
@CajetanRodrigues check.

Reply:
What's to be checked? Its the same api specified in the documentation

Reply:
> @CajetanRodrigues Login using a recently registered user is not working.

I works fine, you can check the gif for the demonstration. When the user signs up, he gets an email with his username and password, which he can then use to login. I have clearly demonstrated that in the gif above. Also, if the user registration is successful, a snackbar pops up at the bottom

Reply:
Working for all test cases now